### PR TITLE
Add options to the user.Count method

### LINKED ISSUE #16269: User count is incorrect
**Expected behavior and actual behavior:**
Expected: The number of users is counted correctly
Actual: The number of users is incorrectly counted

**Steps to reproduce the problem:**
Go to users page
WechatIMG79

**Versions:**
- harbor version: v2.5.0-70155172

### PR DESCRIPTION
**Pull request by stonezdj**
Exclude the admin account by default
  Fixes #16269

Signed-off-by: stonezdj <stonezdj@gmail.com>

### CODE REVIEW

File: src/pkg/user/manager.go [hunk 1]
```diff
@@ -116,7 +116,11 @@ func (m *manager) MatchLocalPassword(ctx context.Context, usernameOrEmail, passw
 	return nil, nil
 }
 
-func (m *manager) Count(ctx context.Context, query *q.Query) (int64, error) {
+func (m *manager) Count(ctx context.Context, query *q.Query, options ...models.Option) (int64, error) {
```
Comment (wy65701436) [inline src/pkg/user/manager.go:119]: should we keep the list & count as same logic?

Comment (stonezdj) [reply src/pkg/user/manager.go:119]: update

File: src/pkg/user/manager.go [hunk 2]
```diff
@@ -116,7 +116,11 @@ func (m *manager) MatchLocalPassword(ctx context.Context, usernameOrEmail, passw
...
+	opts := models.NewOptions(options...)
+	if !opts.IncludeDefaultAdmin && query != nil && query.Keywords != nil {
```
Comment (ywk253100) [inline src/pkg/user/manager.go:121]: If the opts.IncludeDefaultAdmin is false and the query is nil, seems this doesn't work as expected.
How about changing it as following?
```go
if !opts.IncludeDefaultAdmin {
  if query == nil {
    // new a new query
 }
if query.Keywords == nil {
  // new keywords
}
}
```

Comment (stonezdj) [reply src/pkg/user/manager.go:121]: ok, updated

**Review by ywk253100 — APPROVED** [review]

**Review by wy65701436 — APPROVED** [review]
lgtm

**Pull request merged**